build/media-suite_helper.sh

```diff
@@ -1387,7 +1404,18 @@ create_build_dir() {
         fi
     fi
     [[ ! -d $build_dir ]] && mkdir "$build_dir"
-    cd_safe "$build_dir"
+    $nocd || cd_safe "$build_dir"
+    if $print_build_dir; then
+        printf '%s' "$(
+            cd_safe "$(dirname "$build_dir")"
+            pwd -P
+        )"
+        if [[ $build_dir == "$(basename "$PWD")" ]]; then
```

Comment:
What does this if-else do? Only the above printf is showing up for me.

Reply:
That's to filter out when running in the build directory (to prevent build-64bit/build-64bit)

Reply:
That doesn't happen because of the code above.
And actually, the code never works unless it's run from /build because get_first_subdir only returns the dirname, not the full path. Both the do_not_clean and do_not_reconfigure checks don't work.

Reply:
Hmm, it seems I need to relook at all of the code